File: bench/mandel.c
/* Floating-point benchmark: Mandelbrot escape-time. Matches mandel.zeph/.rs.
   Build with -ffp-contract=off so no FMA fusion perturbs the checksum. */
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv) {
    long W = argc > 1 ? atol(argv[1]) : 900;
    long H = W;
    int MAXIT = 256;

    long long total = 0;
    for (long py = 0; py < H; py++) {
        double y0 = (double)py / (double)H * 2.0 - 1.0;
        for (long px = 0; px < W; px++) {
            double x0 = (double)px / (double)W * 3.5 - 2.5;
            double x = 0.0, y = 0.0;
            int it = 0;
            while (x * x + y * y <= 4.0 && it < MAXIT) {
                double xt = x * x - y * y + x0;
                y = 2.0 * x * y + y0;
                x = xt;
                it++;
            }
            total += it;
        }
    }
    printf("%lld\n", total);
    return 0;
}

File: bench/mandel.rs
// Floating-point benchmark: Mandelbrot escape-time. Matches mandel.zeph/.c.
// rustc does not contract to FMA by default, so the checksum matches C's
// -ffp-contract=off build exactly.
use std::env;

fn main() {
    let a: Vec<String> = env::args().collect();
    let w: i64 = if a.len() > 1 { a[1].parse().unwrap() } else { 900 };
    let h = w;
    let maxit = 256;

    let mut total: i64 = 0;
    for py in 0..h {
        let y0 = py as f64 / h as f64 * 2.0 - 1.0;
        for px in 0..w {
            let x0 = px as f64 / w as f64 * 3.5 - 2.5;
            let (mut x, mut y) = (0.0f64, 0.0f64);
            let mut it = 0;
            while x * x + y * y <= 4.0 && it < maxit {
                let xt = x * x - y * y + x0;
                y = 2.0 * x * y + y0;
                x = xt;
                it += 1;
            }
            total += it;
        }
    }
    println!("{}", total);
}
